ts/src/provider.ts

```diff
@@ -194,12 +192,10 @@ export default class Provider {
       )
     ).blockhash;
 
-    await this.wallet.signTransaction(tx);
```

Comment:
The challenge with this change is that the API for simulating is now different from the API for sending a transaction.

Instead, we should turn `Provider` into an interface and then one can use different providers for this specific use case where one doesn't want the provider to automatically sign the simulation.

Reply:
Ok, can you give a bit more insight as to how would the provider switch off signing, would the `simulate` check for something? Or would `signTransaction` do nothing with this specific provider? It is also odd because what would happen when someone provides `signers` but signing is turned off.

Reply:
You would switch off signing by implementing the interface with a new class, which does exactly what the new code here does.

So at some point we want to do

```
export interface Provider {..}
export class HttpProvider implements Provider {...}
export class WsProvider implements Provider {...}
export class UnsignedProvider implements Provider {...}
```

Where `UnsignedProvider` does what the changes here do. That is, it doesn't have a wallet and so doesn't try to sign transactions with it (though it will still have `signers?` optionally passed in, which will sign).